en/components/navdrawer.md

```diff
@@ -316,6 +316,65 @@ After all the steps above are completed, your app should look like that:
 </code-view>
 
 
+<div class="divider--half"></div>
+
+## Hierarchical Navigation
+
+To create a multi-level hierarchical navigation using the `IgxNavigationDrawerComponent`, you can use the `[IgxTreeComponent](tree.md)` in the `igxDrawer` template. The tree can be constructed directly from your application `Routes` object. Here's how this can be acieved:
+
+```html
+<igx-nav-drawer [isOpen]="true" [enableGestures]="true" width="280px">
+    <ng-template igxDrawer>
+        <igx-tree>
+            <igx-tree-node *ngFor="let route of routes">
+                <a igxTreeNodeLink [routerLink]="route.path">{{ route.data?.displayName }}</a>
+                <igx-tree-node *ngFor="let child of route.children">
+                    <a igxTreeNodeLink [routerLink]="[route.path, child.path]">{{ child.data?.displayName }}</a>
+                </igx-tree-node>
+            </igx-tree-node>
+        </igx-tree>
+    </ng-template>
+</igx-nav-drawer>
+```
+
+> [!NOTE]
+> In this example, we're not using the `igxDrawerItem`, instead we're populating directly with custom `igxDrawer` content, in this case using an `igx-tree`.
+
+```typescript
+import { menusRoutes } from '../../menus-routing.module';
+
+@Component({
+  selector: 'app-nav-drawer-hierarchical',
+  templateUrl: './nav-drawer-hierarchical.component.html',
+  styleUrls: ['./nav-drawer-hierarchical.component.scss']
+})
+export class NavDrawerHierarchicalComponent {
+    public routes = menusRoutes;
+}
+```
+
+In this example, we're populating the routes with custom routing `data`, which contains a `displayName` property, used to visualize the link text in the `igx-tree` nodes. A sample `Route` looks like this:
+
+```typescript
+export const menusRoutes: Routes = [
+    {
+        component: NavDrawerHierarchicalComponent,
+        path: 'navigation-drawer-hierarchical',
+        data: { displayName: 'Hierachical Drawer Menu' }
+    }
+];
+```
+
+There's also child routing extracted from the `children` property of the routes. The sample shows two levels of hierachy, but if your routing has more, then all you need to do is define the levels below in the tree node templates.
```

Comment:
I was going to ask about this when I saw the snippet above first. Much needed comment.
Since this isn't in the sample below from what I see, if there's a suitable section in the Tree topic (assuming there is) - maybe make the `"tree node templates"` bit a link to that part for guidance on how to make use of templates

Reply:
@damyanpetev "Below" in this case doesn't refer to the "content of the topic", but to the levels in the hierarchy you're creating a tree for.

Reply:
Oh no no, I didn't mean `below` in the sentence, but the sample embedded in the topic content after this and that it's not using templates but the snipped from the topic, which is perfectly fine. All I asked was if it'd be better if we link directly to the Tree topic when mentioning the template approach (for readers not familiar with that) - just make that a link to https://www.infragistics.com/products/ignite-ui-angular/angular/components/tree#templating directly?

Reply:
@damyanpetev I could do that, but I'm not certain if it's necessary. The templating section is in the same topic and is above the current section. There's also navigation item in the right-side navigation for it.